### PR TITLE
Revise glossary entry for kube-proxy

### PR DESCRIPTION
**Pull request by sftim**
- Mark as relevant to networking.
- Unmark as a core object. kube-proxy is not an API object.
- Fix hyperlink to reference docs.
- Include hyperlink in definition. The definition is used in [/docs/concepts/overview/components/](https://kubernetes.io/docs/concepts/overview/components/) and the hyperlink is particularly useful to have on that page.
- Revise wording.

/priority backlog

### CODE REVIEW

File: content/en/docs/reference/glossary/kube-proxy.md [hunk 1]
```diff
@@ -2,21 +2,21 @@
...
 
-It enables the Kubernetes service abstraction by maintaining network rules on
-the host and performing connection forwarding.
-
-<!--more--> 
-
-`kube-proxy` is responsible for request forwarding. `kube-proxy` allows TCP and UDP stream forwarding or round robin TCP and UDP forwarding across a set of backend functions.
+<!--more-->
 
+kube-proxy maintains network rules on nodes so that sessions (that
```
Comment (mistyhacks) [inline content/en/docs/reference/glossary/kube-proxy.md]: I think this can be improved by more active voice:

kube-proxy maintains network rules on nodes, such as `iptables` rules on Linux nodes or <what is this called on Windows nodes??> on Windows Server nodes. These network rules allow network communication to your Pods from network sessions inside or outside of your cluster.

Comment (sftim) [reply content/en/docs/reference/glossary/kube-proxy.md]: That's good feedback @mistyhacks. I think the wording I proposed is not strictly wrong but it would be misleading, especially for less common cluster configutations. On Windows nodes (and Linux nodes where iptables isn't available) I _think_ `kube-proxy` is a userland proxy for TCP & SCTP. For UDP, `kube-proxy` in userland mode acts as a rewriting UDP forwarder.
There are acceleration options on Linux and, AFAIK, just Linux.

Sometimes kube-proxy manages the list of rules *and* implements the forwarding itself. Sometimes, kube-proxy gets OS kernel help: it manages the list of rules and lets another component do the forwarding ([iptables mode](https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-iptables) / [ipvs mode](https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-ipvs)). iptables mode and ipvs mode are more efficient than userland forwarding.

Also relevant:
- issue #14770
- https://kubernetes.io/docs/concepts/cluster-administration/proxies/
- https://kubernetes.io/docs/concepts/services-networking/service/#proxy-mode-userspace and below
  _definitely deserves a rewrite_

**Comment by sftim** [comment]
Marked this as WIP. To find the right new wording, I need to take more care than I'd realized.

File: content/en/docs/reference/glossary/kube-proxy.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/reference/glossary/kube-proxy.md]: Maybe “an internal set of traffic forwarding rules” would work for Windows(?)

Comment (sftim) [reply content/en/docs/reference/glossary/kube-proxy.md]: How about this rewording?

File: content/en/docs/reference/glossary/kube-proxy.md [hunk 2]
```diff
@@ -2,21 +2,25 @@
...
 
-<!--more--> 
-
-`kube-proxy` is responsible for request forwarding. `kube-proxy` allows TCP and UDP stream forwarding or round robin TCP and UDP forwarding across a set of backend functions.
+kube-proxy maintains network rules on nodes. These network rules allow
+network communication to your Pods from network sessions inside or outside
+of your cluster.
 
+kube-proxy can optionally copy those rules to the operating system's packet
+filtering layer (eg iptables on Linux). Using the OS' packet filtering layer
```
Comment (mistyhacks) [inline content/en/docs/reference/glossary/kube-proxy.md]: consider avoiding Latin (eg). Also you should put iptables in backticks. Also don't use plural possessive for "OS'". You could reword this to "To improve performance, kube-proxy can optionally....." But can you provide some info about when it would or wouldn't improve performance? Presumably there is a trade-off since it's not enabled by default.

Comment (sftim) [reply content/en/docs/reference/glossary/kube-proxy.md]: kube-proxy uses the operating system packet filtering layer if there is one and it's available. I'll say that.

**Comment by mistyhacks** [comment]
/lgtm
/approve
/hold

You can lift the hold after you've made the final change. Thanks!

**Comment by sftim** [comment]
(also needs an `/lgtm` I think)

**Comment by mistyhacks** [comment]
It's a funny quirk but you an supply that LGTM, as the author. But I'll do it for you again. SOrry for the churn.

/lgtm

**Comment by sftim** [comment]
Thanks. It feels odd to `/lgtm` my own PR. Noted for next time.
